src/NUnitFramework/framework/Constraints/CollectionOrderedConstraint.cs

```diff
@@ -114,13 +168,33 @@ public CollectionOrderedConstraint By(string propertyName)
         public override string Description
         {
             get 
-            { 
-                string desc = propertyName == null
-                    ? "collection ordered"
-                    : "collection ordered by "+ MsgUtils.FormatValue(propertyName);
+            {
+                string desc;
 
-                if (descending)
-                    desc += ", descending";
+                if (this.properties == null || this.properties.Count == 0)
+                {
+                    desc = "collection ordered";
+
+                    if (descending)
+                        desc += ", descending";
+                }
+                else
+                {
+                    var first = this.properties.ElementAt(0);
+                    desc = "collection ordered by " + MsgUtils.FormatValue(first.Key);
+
+                    if (first.Value)
+                        desc += ", descending";
+
+                    for (int i = 1; i < this.properties.Count; i++)
```

Comment:
@jnsn My research indicates that the order is not guaranteed to be anything in particular, only that the order of keys and values will match one another. This probably needs to be re-implemented independent of whatever we decide for the user interface.